File: code-examples/java-rs/stable-api/options.java
// Import the necessary MongoDB classes
import com.mongodb.ConnectionString;
import com.mongodb.MongoClientSettings;
import com.mongodb.ServerApi;
import com.mongodb.ServerApiVersion;
import com.mongodb.reactivestreams.client.MongoClients;
import com.mongodb.reactivestreams.client.MongoClient;

// Replace the placeholders in the connection string uri with your credentials
String uri = "mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority";

//Create ServerApi settings for version 1, strict mode, and deprecation errors
ServerApi serverApi = ServerApi.builder()
   .version(ServerApiVersion.V1)
   .strict(true)
   .deprecationErrors(true)
   .build();

// Create a Mongo client settings with the uri and serverApi settings
MongoClientSettings settings = MongoClientSettings.builder()
    .applyConnectionString(new ConnectionString(uri))
    .serverApi(serverApi)
    .build();

// Create a client using the settings
MongoClient client = MongoClients.create(settings);

File: code-examples/pymongo/stable-api/options.py
# Stable API - with options

# begin serverApiVersion
from pymongo import MongoClient, WriteConcern

# Replace the placeholders in the connection string uri with your credentials
uri = "mongodb+srv://<user>:<password>@<cluster-url>"

# Create a client with options to specify Stable API Version 1, return
# errors for commands outside of the API version, and raise exceptions
# for deprecated commands
client = MongoClient(uri, serverSelectionTimeoutMS=5000, serverApi="1", writeConcern=WriteConcern(w="majority"))
# end serverApiVersion